web-common/src/runtime-client/invalidation.ts

```diff
@@ -98,11 +98,20 @@ export function invalidationForMetricsViewData(query, metricsViewName: string) {
   );
 }
 
-export function isProfilingQuery(queryHash: string, name: string) {
-  const r = new RegExp(
-    `/v1/instances/[a-zA-Z0-9-]+/queries/[a-zA-Z0-9-]+/tables/${name}`
+const ProfilingQueryExtractor =
+  /v1\/instances\/[a-zA-Z0-9-]+\/queries\/([a-zA-Z0-9-]+)\/tables\/(.+?)"/;
+export function isProfilingQuery(
+  queryHash: string,
+  name: string,
+  ignoreProfileColumns = false
+) {
+  const queryExtractorMatch = ProfilingQueryExtractor.exec(queryHash);
+  if (!queryExtractorMatch) return false;
+  // TODO: move the query matching to a separate module and reuse in http queue
+  const [, type, table] = queryExtractorMatch;
+  return (
+    table === name && (!ignoreProfileColumns || type !== "columns-profile")
   );
```

Comment:
This is hard to follow. Could it be refactored to contain `isTableProfileQuery` and `isColumnProfileQuery`, or something along those lines?